### PR TITLE
Wikidata values for catalog providers

### LINKED ISSUE #418: Dump Wikidata values that are not in the target catalog
Goal: submit extra data to target catalog providers.
Every validation criterion should also output Wikidata values not available in the given target catalog.

### PR DESCRIPTION
**Pull request by marfox**
This PR closes #418 .
Please bear in mind that relevant refactoring suggestions from #419 are not yet addressed here: we have opened issues accordingly (#420 #421 #422) and will tackle them as soon as all the new validator features are implemented.

### CODE REVIEW

**Comment by marfox** [comment]
@e-dorigatti : friendly ping

File: soweego/validator/checks.py [hunk 1]
```diff
@@ -540,27 +582,25 @@ def links(
...
+      6. ``list`` of URLs found in Wikidata but not in the target catalog
+      7. ``dict`` of links gathered from Wikidata
 
     """
     # Target catalog side first:
     # enable early return in case of no target links
     target_links = data_gathering.gather_target_links(entity, catalog)
     if target_links is None:
-        return None, None, None, None, None, None
+        return None, None, None, None, None, None, None
```
Comment (e-dorigatti) [inline soweego/validator/checks.py:600]: The declared return type does not match what you return here, so you need to change either one of them:

 - The return type it should be `Tuple[Optional[defaultdict], Optional[list], (etc)]]`
 - Or change this to `return Non`

PS this is caught by [mypy](http://mypy-lang.org/), aren't you using it?
```
checks.py:600: error: Incompatible return value type (got "Tuple[None, None, None, None, None, None, None]", expected "Optional[Tuple[defaultdict[Any, Any], List[Any], List[Any], List[Any], List[Any], List[Any], Dict[Any, Any]]]")
```

File: soweego/validator/checks.py [hunk 2]
```diff
@@ -649,25 +706,23 @@ def bio(
...
+      4. ``generator`` of statements found in Wikidata but not in the target catalog
+      5. ``dict`` of biographical data gathered from Wikidata
 
     """
     # Target catalog side first:
     # enable early return in case of no target data
     target_bio = data_gathering.gather_target_biodata(entity, catalog)
     if target_bio is None:
-        return None, None, None, None
+        return None, None, None, None, None
```
Comment (e-dorigatti) [inline soweego/validator/checks.py:722]: Same comment as above regarding return types

**Review by e-dorigatti — APPROVED** [review]
sorry for the delay! code is fine but return values need to be made consistent with type annotations. you may look into [pre-commit](https://pre-commit.com/) to automatically run type checking, linting, formatting, etc. on commit

File: soweego/validator/checks.py (hunk 1 above)
Comment (marfox) [reply soweego/validator/checks.py:600]: > PS this is caught by mypy, aren't you using it?

To cut a long story short: when we started the project, type hints were relatively new. We decided to add them just to relevant public functions, hence no automatic check was implemented.
Later, we put a script in place, and we should definitely automate its execution: https://github.com/Wikidata/soweego/blob/9ed8b4e28fd2ea4150b37cbb49b44ce8bfb2c381/scripts/enforce_style.sh

**Comment by marfox** [comment]
Many thanks for the review, your feedback is always super constructive and much appreciated.

> return values need to be made consistent with type annotations.

Fixed.

> you may look into [pre-commit](https://pre-commit.com/) to automatically run type checking, linting, formatting, etc. on commit

Very interesting suggestion! We currently have a similar mechanism in place on _Travis CI_, see [this dotfile](https://github.com/Wikidata/soweego/blob/9ed8b4e28fd2ea4150b37cbb49b44ce8bfb2c381/.travis.yml).
However, it's a bit hacky and doesn't run _mypy_ type hints check.
From a first glimpse, it looks like  _pre-commit_ is a much better solution:
- it can cater for CI , see https://pre-commit.com/#usage-in-continuous-integration;
- no hack seems required to autofix PRs, see https://pre-commit.com/#pre-commitci-example;
- it's claimed to be faster than Travis, see https://pre-commit.com/#pre-commitci-example.

**Pull request merged**